lib/topological_inventory/openshift/collectors_pool.rb

```diff
@@ -0,0 +1,85 @@
+require "config"
+require "topological_inventory/openshift/collector"
+require "topological_inventory/openshift/logging"
+
+module TopologicalInventory::Openshift
+  class CollectorsPool
+    include Logging
+
+    def initialize(config_name, metrics, poll_time: 10)
+      self.collectors  = {}
+      self.collector_threads = {}
+      self.config_name = config_name
+      self.metrics     = metrics
+      self.poll_time   = poll_time
+    end
+
+    def run!
+      loop do
+        reload_config
+
+        remove_old_collectors
+        add_new_collectors
+
+        sleep(poll_time)
+      end
+    end
+
+    def stop!
+      collectors.each_value { |collector| collector.stop }
+
+      # Wait for end of collectors to ensure metrics are stopped after them
+      collector_threads.each { |thread| thread.join }
+    end
+
+    private
+
+    attr_accessor :collectors, :collector_threads, :config_name, :metrics, :poll_time
+
+    def reload_config
+      clear_settings
```

Comment:
Can you just use `Settings.reload!` here?

Reply:
> > bin/openshift-multicollector
> 
> Hey no problem if you're still keeping the old one for testing, but I'm fine with just drop-in replacing the old collector and not keeping both around.

it would require that all type of supported collectors and orchestrator will deploy changes at the same time, I suggest to remove it after orchestrator will be changed and deployed too

Reply:
> Can you just use `Settings.reload!` here?
I experienced some problems before when reloading files, I'll look at it in more detail

Reply:
I remember where the problem is - when using `Config.load_and_set_settings` - values to `Settings` are added. `Settings.reload!` doesn't clear previously loaded values. In current situation when there is only one root key it is probably ok, but `clear_settings` is definitely safer

Reply:
..and there is no method `Config.clear_and_load`

Reply:
Hm if that's true that seems like an issue with the config gem, @Fryguy are you aware of this?

Reply:
@agrare I think this is intention https://github.com/railsconfig/config/blob/master/lib/config.rb#L40. Issue is that `Config.clear!` method is missing

Reply:
@agrare sorry, false alarm. This is behaviour for `load_file` only, `load_and_set_settings` clears previous Settings. It doesn't remove values only when more files in one call are passed